shoes-core/lib/shoes/builtin_methods.rb

```diff
@@ -37,6 +37,13 @@ def error(message = '')
       Shoes.logger.error message
     end
 
+    def exit(message = '')
+      Shoes::LOG << ['exit', message]
+      for app in Shoes.apps
```

Comment:
It's a minor point, but you rarely see Ruby code actually using `for`. The more common expression for this would be `Shoes.apps.each do |app|`.